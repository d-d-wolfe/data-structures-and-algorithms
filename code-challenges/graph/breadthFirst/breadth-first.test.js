'use strict';

const Graph = require('../graph.js');
const bf = require('./breadth-first.js');

describe('Testing the breadth first graph traversal method', () => {
  it('should return all of the nodes in the graph', () => {
    const graph = new Graph();

    const one = graph.addNode(1);
    const two = graph.addNode(2);
    const four = graph.addNode(4);
    const six = graph.addNode(6);
    const seven = graph.addNode(7);
    const eight = graph.addNode(8);

    graph.addEdge(one, four);
    graph.addEdge(one, two);
    graph.addEdge(two, four);
    graph.addEdge(four, six);
    graph.addEdge(six, eight);
    graph.addEdge(six, seven);

  expect(bf(one)).toMatchSnapshot();
  });

  it('should return all the nodes in the graph from a different starting point', () => {
    const graph = new Graph();

    const one = graph.addNode(1);
    const two = graph.addNode(2);
    const four = graph.addNode(4);
    const six = graph.addNode(6);
    const seven = graph.addNode(7);
    const eight = graph.addNode(8);

    graph.addEdge(one, four);
    graph.addEdge(one, two);
    graph.addEdge(two, four);
    graph.addEdge(four, six);
    graph.addEdge(six, eight);
    graph.addEdge(six, seven);

    expect(bf(four)).toMatchSnapshot();
  });
});
