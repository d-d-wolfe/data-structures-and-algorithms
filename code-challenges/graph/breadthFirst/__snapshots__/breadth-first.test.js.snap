// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the breadth first graph traversal method should return all of the nodes in the graph 1`] = `
Set {
  Vertex {
    "value": 1,
  },
  undefined,
}
`;

exports[`Testing the breadth first graph traversal method should return all the nodes in the graph from a different starting point 1`] = `
Set {
  Vertex {
    "value": 4,
  },
  undefined,
}
`;
